pkg/feature/servicemesh/conditions.go

```diff
@@ -90,6 +90,9 @@ func WaitForControlPlaneToBeReady(ctx context.Context, cli client.Client, f *fea
 
 	return wait.PollUntilContextTimeout(ctx, interval, duration, false, func(ctx context.Context) (bool, error) {
 		ready, err := CheckControlPlaneComponentReadiness(ctx, cli, smcp, smcpNs)
+		if k8serr.IsNotFound(err) {
```

Comment:
i bet this is the real fix for the case
when smcp is not even created by dsci yet, but kserve has started check on it.

Reply:
The mojority of the cases yes, but also the check on the presence of the expected status properties was causing lot of failures